src/detect-cipservice.c

```diff
@@ -157,8 +157,9 @@ static DetectCipServiceData *DetectCipServiceParse(const char *rulestrc)
             goto error;
         }
 
-        sscanf(token, "%2" SCNu8, &var);
-        input[i++] = var;
+        if (sscanf(token, "%2" SCNu8, &var) == 1) {
```

Comment:
what does it mean if sscanf returns another value? What should we do with the token then?

Reply:
Looks right as done a few lines above

Reply:
OK! Thank you :).

I am also curious, I tried to search what `%2` formats to, but I could not find any answers.

Reply:
`"%2" SCNu8` 

because `SCNu8` is a portable macro to scan a uint8_t, so it would look like `"%2c"` in the end

Reply:
yes please